common/src/main/java/com/wynntils/sockets/model/SocketModel.java

```diff
@@ -0,0 +1,202 @@
+    public static void onDisconnect(SocketEvent.Disconnected event) {
+        pingScheduler.shutdown();
+    }
+
+    private static void sendPing() {
+        if (!isSocketOpen()) return;
+
+        hadesConnection.sendPacketAndFlush(new HCPacketPing(System.currentTimeMillis()));
+    }
+
+    @SubscribeEvent
+    public static void onFriendListUpdate(RelationsUpdateEvent.FriendList event) {
+        if (!SocketFeature.INSTANCE.shareWithFriends || !isSocketOpen()) return;
+
+        hadesConnection.sendPacket(new HCPacketSocialUpdate(
+                event.getChangedPlayers().stream().toList(),
+                event.getChangeType().getPacketAction(),
+                SocialType.FRIEND));
+    }
+
+    @SubscribeEvent
+    public static void onPartyListUpdate(RelationsUpdateEvent.PartyList event) {
+        if (!SocketFeature.INSTANCE.shareWithParty || !isSocketOpen()) return;
+
+        hadesConnection.sendPacket(new HCPacketSocialUpdate(
+                event.getChangedPlayers().stream().toList(),
+                event.getChangeType().getPacketAction(),
+                SocialType.PARTY));
+    }
+
+    @SubscribeEvent
+    public static void onWorldStateChange(WorldStateEvent event) {
+        if (!isSocketOpen()) return;
+
+        resendWorldData();
+    }
+
+    @SubscribeEvent
+    public static void onClassChange(CharacterUpdateEvent event) {
+        if (!isSocketOpen()) return;
+
+        resendWorldData();
+    }
+
+    @SubscribeEvent
+    public static void onTick(ClientTickEvent.End event) {
+        if (!isSocketOpen()) return;
+        if (!WorldStateManager.onWorld()) return;
+        if (!SocketFeature.INSTANCE.shareWithParty
+                && !SocketFeature.INSTANCE.shareWithGuild
+                && !SocketFeature.INSTANCE.shareWithFriends) return;
+
+        tickCountUntilUpdate--;
+
+        if (tickCountUntilUpdate <= 0) {
+            LocalPlayer player = McUtils.player();
+
+            float pX = (float) player.getX();
+            float pY = (float) player.getY();
+            float pZ = (float) player.getZ();
+
+            if (lastSentStatus != null
+                    && lastSentStatus.equals(
+                            pX,
+                            pY,
+                            pZ,
+                            ActionBarModel.getCurrentHealth(),
+                            ActionBarModel.getMaxHealth(),
+                            ActionBarModel.getCurrentMana(),
+                            ActionBarModel.getMaxMana())) {
+                tickCountUntilUpdate = 1;
+                return;
+            }
+
+            tickCountUntilUpdate = TICKS_PER_UPDATE;
+
+            lastSentStatus = new PlayerStatus(
+                    pX,
+                    pY,
+                    pZ,
+                    ActionBarModel.getCurrentHealth(),
+                    ActionBarModel.getMaxHealth(),
+                    ActionBarModel.getCurrentMana(),
+                    ActionBarModel.getMaxMana());
```

Comment:
Can't this and the class `PlayerStatus` just be removed and inlined?

Reply:
We have a class called PlayerStatus?

Reply:
A record